### PR TITLE
SCALRCORE-19877 IaCP Provider > VCS providers > Scalr VCS provider resource

### PR DESCRIPTION
**Pull request by penja**
### Changelog
* [ ] I have updated CHANGELOG.md according to [CONTRIBUTING.md](../CONTRIBUTING.md)

### Documentation
* [x] I have updated resource/data source documentation in [docs](../docs)

### State
* [ ] My changes affect the state
* [ ] I have included a state migration and a unit test for it

### CODE REVIEW

File: docs/resources/scalr_vcs_provider.md [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+```
+
+## Argument Reference
+
+* `name` - (Required) Name of the vcs provider.
+* `vcs_type` (Required) The vcs provider type is one of `github`, `github_enterprice`, `gitlab`, `github_enterprice`, `bitbucket_enterprise`. 
+   The other providers currently is not supported in resource.
+* `token` (Required) The personal access token for provider
+  * Github token can be generated by url https://github.com/settings/tokens/new?description=example-vcs-resouce&scopes=repo
+  * Gitlab token can be generated by url https://gitlab.com/-/profile/personal_access_tokens?name=example-vcs-resouce&scopes=api,read_user,read_registry
```
Comment (brezitskiy) [inline docs/resources/scalr_vcs_provider.md:33]: @penja Let's add the instructions for bitbucket as well

File: scalr/resource_scalr_vcs_provider.go [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+	scalrClient := meta.(*scalr.Client)
+	// Get attributes.
+	name := d.Get("name").(string)
+	token := d.Get("token").(string)
+	vcsType := scalr.VcsType(d.Get("vcs_type").(string))
+	options := scalr.VcsProviderCreateOptions{
+		Name:     &name,
+		VcsType:  vcsType,
+		Token:    token,
+		AuthType: "personal_token",
```
Comment (brezitskiy) [inline scalr/resource_scalr_vcs_provider.go:71]: @penja can't we use the constant introduced in client here?

File: scalr/resource_scalr_vcs_provider.go [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+	d.SetId(provider.ID)
+
+	return resourceScalrVcsProviderRead(d, meta)
+}
+
+func resourceScalrVcsProviderRead(d *schema.ResourceData, meta interface{}) error {
+	scalrClient := meta.(*scalr.Client)
+	providerID := d.Id()
+
+	log.Printf("[DEBUG] Read provider with ID: %s", providerID)
```
Comment (brezitskiy) [inline scalr/resource_scalr_vcs_provider.go:100]:
```suggestion
	log.Printf("[DEBUG] Read vcs provider with ID: %s", providerID)
```

**Review by brezitskiy — APPROVED** [review]
LGTM, just a few minor notes

File: scalr/resource_scalr_vcs_provider.go (hunk 2 above)
Comment (penja) [reply scalr/resource_scalr_vcs_provider.go:71]: @brezitskiy yes we can

File: docs/resources/scalr_vcs_provider.md (hunk 1 above)
Comment (penja) [reply docs/resources/scalr_vcs_provider.md:33]: 🙈

**Pull request merged**